Documentation for `cfssl genkey -initca csrjson` is incorrect

The docs say:

```
Generating self-signed root CA certificate and private key

cfssl genkey -initca csrjson | cfssljson -bare ca
To generate a self-signed root CA certificate, specify the key request as the JSON file in the same format as in 'genkey'. Three PEM-encoded entities will appear in the output: the private key, the csr, and the self-signed certificate.
```

But it only generates two things:

```
$ cfssl gencert -initca $c_json                     
2015/03/21 18:13:45 [ERROR] open ca-key.pem: no such file or directory
2015/03/21 18:13:45 [INFO] generating a new CA key and certificate from CSR
2015/03/21 18:13:45 [INFO] generate received request
2015/03/21 18:13:45 [INFO] received CSR
2015/03/21 18:13:45 [INFO] generating key: rsa-2048
2015/03/21 18:13:46 [INFO] encoded CSR
2015/03/21 18:13:47 [INFO] signed certificate with serial number 7336380156049125026
{"cert":"-----BEGIN CERTIFICATE-----\nMIID7jCCAtigAwIBAgIIZdAO4dKIbqIwCwYJKoZIhvcNAQELMIGEMQswCQYDVQQG\nEwJBVTEWMBQGA1UEChMNQmV2cnkgUHR5IEx0ZDEeMBwGA1UECxMVQmV2cnkgTGVh\nZGVyc2hpcCBUZWFtMQ4wDAYDVQQHEwVQZXJ0aDEaMBgGA1UECBMRV2VzdGVybiBB\ndXN0cmFsaWExETAPBgNVBAMTCGJldnJ5Lm1lMB4XDTE1MDMyMTEwMDkwMFoXDTIw\nMDMxOTEwMDkwMFowgYQxCzAJBgNVBAYTAkFVMRYwFAYDVQQKEw1CZXZyeSBQdHkg\nTHRkMR4wHAYDVQQLExVCZXZyeSBMZWFkZXJzaGlwIFRlYW0xDjAMBgNVBAcTBVBl\ncnRoMRowGAYDVQQIExFXZXN0ZXJuIEF1c3RyYWxpYTERMA8GA1UEAxMIYmV2cnku\nbWUwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQD5TSyYY/22DHF8DWAW\ngzvLzd8Wlym5QuJ8YQ6aamONfYxfaaIqunxJ39wFXHNyXNRYwLK8x81YjOq8GO8o\nspJ6QmUjJaSTOAccR3JuIf5a/xiC2U5YbgNm8blpPQwIOTuF/N3RUOhucQwV69WZ\nsA5nTJ/ir9re0HMruPtSVfE1xCfYikDCn4w69iLS6/DAT3dw2vWnTT24G1s3oEot\n2DawpGDzUREbKMysALc5VOiiWPE3lj90c/O87bj1iRfqp/C5OLopOFgimOpzcFiI\ntnCTnWD9/HH2Ky1cQrG/FLX+fk2y/wGZXZkBgf9YsJfck1G8zizvBJT2lFEew0TA\nNH0DAgMBAAGjZjBkMA4GA1UdDwEB/wQEAwIABjASBgNVHRMBAf8ECDAGAQH/AgEC\nMB0GA1UdDgQWBBQU2/pm/e0EJYClmYbEzWTVnZQHdTAfBgNVHSMEGDAWgBQU2/pm\n/e0EJYClmYbEzWTVnZQHdTALBgkqhkiG9w0BAQsDggEBAF1t4ZRJqnAIKHF1ZguP\ns2dTOoJIm1P0CaX1HKu2rHGrCqFpJLGcpW3lvzVuvvL0WCs+sPUWvJ96p4XEiLWX\nCSw+ZQuDGvkJRQf+43WkEr53E7bADkon/cji/iVjI761AcHLX4WryDhLxff/Vxa0\nylZh9Dx5PDvMbhCUgQfSnSdZZWQuFRiNR6TItsPQ/wV3/mtyvlczJOnEovyAHGTq\ntTBD69ZqS1PQMh8pawi4W9jk7MKznNlzN3o5HHqabK3HlKvMVlHF9fpq2YcFDEmh\nh+r5B+agWxg+asK8ILtG7LAIgKexMSrXrjSPePgJWs3pyxJ32VCOntRcv58r3PNb\nBgM=\n-----END CERTIFICATE-----\n","key":"-----BEGIN RSA PRIVATE KEY-----\nMIIEogIBAAKCAQEA+U0smGP9tgxxfA1gFoM7y83fFpcpuULifGEOmmpjjX2MX2mi\nKrp8Sd/cBVxzclzUWMCyvMfNWIzqvBjvKLKSekJlIyWkkzgHHEdybiH+Wv8YgtlO\nWG4DZvG5aT0MCDk7hfzd0VDobnEMFevVmbAOZ0yf4q/a3tBzK7j7UlXxNcQn2IpA\nwp+MOvYi0uvwwE93cNr1p009uBtbN6BKLdg2sKRg81ERGyjMrAC3OVTooljxN5Y/\ndHPzvO249YkX6qfwuTi6KThYIpjqc3BYiLZwk51g/fxx9istXEKxvxS1/n5Nsv8B\nmV2ZAYH/WLCX3JNRvM4s7wSU9pRRHsNEwDR9AwIDAQABAoIBACywbOIUapKLwHwg\nWa0vcTSjfu4pwsz29Gqfa2K614Wxj8w6v1sxer4zZtl/LSK/CUGvqVq15lWNTTbS\nsWx2NlO8Oy0Fvm61iHDFTfn4FgRx3g72rVikv+9gPYBj9Rg0SjtGjOzMuB20Hhp/\n4BZgvPQ/hBm6tF1kuICJ9GNaDJc6a+eP4Vo58ugZ3m8WEIQ6ay0QMiIyOvSx/r44\ntegD2NAQr5DWywS03ZL9MBujgpMapfbacRbbnbjjTS/9IvjygMjX68EAJ+20uuL4\nQqQ9vGE9WUoXD9ekzdVEjuKjDXd5/lCW8Kv3e0yI3QQnYjCG/DGvoqSaPBFjKppw\nmAbSq8ECgYEA/HQCaahcUsiYRKwygavgRHWMNE2JuUvYV5TadkkW4ZRQXx2dgp6N\n23W3Z9Cns72F9wNmBVNNskYwuPX0mI84g8vhTf67Y4vdJfiDViSXqFBH9IvXVdav\nb7dGvu/iTwgQp6Bzy3iiD9p+OucW8QCSZkH6PZ6iDVknKgTeobHHlZkCgYEA/M3U\nQ1BoEeq5ErCxmQr0694Ix7+md3V6/E8x0iJtz8CDFWnUEmBd4L3EDnQwxpMgAcEf\nlH1wxKJikOFImbjuCWZ72A38ze9HCtPwSVTbvWPO3/M6Ev+93azzfYEkoMjEKc7K\nSl+pVaNk1YF0jl5Jr5VqokGMRD7e0IM0VooNUPsCgYAEuftNlRxdZGoNZxfu+5PQ\nVE8Sr5zzfisHwL6yVj36bTdwAHLxfhxzzmlnpS1unQ7ECClB0kRzIQNKgzhYHt5V\nAV0vg6RogbbbPiLfWwIH4bDnMDMQ5VsVgV3utzkKd8EwEj34wkJuf427dpYkXvle\ntr+DpdOs4k1saPjuGBD/iQKBgFytsLWE0sJOxM6fTFyI8cVOvvVLw2FkBlRdTCx3\nRr61GtlM1Nxvy7rKlwedfQwvf7pk7Ijpe9/6kBwBGzaP32Q+2AQbDKUKpYbIr/yS\nSYSjBIKBjrbMXvTXpORL5fuz3oZdMzgU3eVxY1QKg4p18pmWM2OMmDmkutbCYvue\n3MblAoGAShBFBb4Bq4VK9OF06H3tjzk9+WIIwQP0jD6f4iVLkOM93SUMMMzkSsyq\nzB5oik4tG8rbg/Gk6cjjO1W0Nyvj/azaS0/RXZo3nAjUQZ8hxvIbVkRbc1krgcl5\nzCdk+NUH7HmvxoF7CkDh8+HlAoSd+di0D1Pv6lxMt1GfQfiWurs=\n-----END RSA PRIVATE KEY-----\n"}
```

--- a/api/initca/initca.go
+++ b/api/initca/initca.go
@@ -38,7 +38,7 @@ func initialCAHandler(w http.ResponseWriter, r *http.Request) error {
 		return errors.NewBadRequest(err)
 	}
 
-	key, cert, err := initca.New(req)
+	key, _, cert, err := initca.New(req)
 	if err != nil {
 		log.Warningf("failed to initialise new CA: %v", err)
 		return err

--- a/cli/gencert/gencert.go
+++ b/cli/gencert/gencert.go
@@ -52,7 +52,7 @@ func gencertMain(args []string, c cli.Config) (err error) {
 		if err != nil {
 			log.Errorf("%v\n", err)
 			log.Infof("generating a new CA key and certificate from CSR")
-			cert, key, err = initca.New(&req)
+			cert, _, key, err = initca.New(&req)
 			if err != nil {
 				return
 			}

--- a/cli/genkey/genkey.go
+++ b/cli/genkey/genkey.go
@@ -41,13 +41,13 @@ func genkeyMain(args []string, c cli.Config) (err error) {
 	}
 
 	if c.IsCA {
-		var key, cert []byte
-		cert, key, err = initca.New(&req)
+		var key, csrPEM, cert []byte
+		cert, csrPEM, key, err = initca.New(&req)
 		if err != nil {
 			return
 		}
 
-		cli.PrintCert(key, nil, cert)
+		cli.PrintCert(key, csrPEM, cert)
 	} else {
 		if req.CA != nil {
 			err = errors.New("ca section only permitted in initca")

--- a/initca/initca.go
+++ b/initca/initca.go
@@ -44,7 +44,7 @@ func validator(req *csr.CertificateRequest) error {
 }
 
 // New creates a new root certificate from the certificate request.
-func New(req *csr.CertificateRequest) (cert, key []byte, err error) {
+func New(req *csr.CertificateRequest) (cert, csrPEM, key []byte, err error) {
 	if req.CA != nil {
 		if req.CA.Expiry != "" {
 			CAPolicy.Default.ExpiryString = req.CA.Expiry
@@ -57,7 +57,7 @@ func New(req *csr.CertificateRequest) (cert, key []byte, err error) {
 	}
 
 	g := &csr.Generator{Validator: validator}
-	csr, key, err := g.ProcessRequest(req)
+	csrPEM, key, err = g.ProcessRequest(req)
 	if err != nil {
 		log.Errorf("failed to process request: %v", err)
 		key = nil
@@ -77,7 +77,7 @@ func New(req *csr.CertificateRequest) (cert, key []byte, err error) {
 	}
 	s.SetPolicy(CAPolicy)
 
-	signReq := signer.SignRequest{Hosts: req.Hosts, Request: string(csr)}
+	signReq := signer.SignRequest{Hosts: req.Hosts, Request: string(csrPEM)}
 	cert, err = s.Sign(signReq)
 
 	return

--- a/initca/initca_test.go
+++ b/initca/initca_test.go
@@ -93,7 +93,7 @@ func TestInitCA(t *testing.T) {
 				Size: param.keyLen,
 			},
 		}
-		certBytes, keyBytes, err := New(req)
+		certBytes, _, keyBytes, err := New(req)
 		if err != nil {
 			t.Fatal("InitCA failed:", err)
 		}
@@ -190,7 +190,7 @@ func TestInvalidCryptoParams(t *testing.T) {
 				Size: test.keyLen,
 			},
 		}
-		_, _, err := New(req)
+		_, _, _, err := New(req)
 		if err == nil {
 			t.Fatal("InitCA with bad params should fail:", err)
 		}

